### DISCUSSION TITLE
How to use Open VSLAM output to perform path planning

### DISCUSSION THREAD

**Discussion by Udaykiran87** [stella-cv/stella_vslam#117] [Q&A] [answered]
Hello,

I would like to use Open VSLAM output to perform path planing for my robotic application. Ex: using dijkstra algorithm.
But i am not sure where to start. To be specific, which Open VSLAM outputs could be useful for path planning task? I mean how can i obtain free space and occupied space from the map etc. Any help or overview would be highly appreciated. 

Thanks and Regards,
Udaykiran Patnaik.

**Comment by mirellameelo** [comment, marked as answer]
You'd need a map like an occupancy grid map (OGM). For now, Openvslam outputs a sparse map of the environment. There are intentions to develop an output denser,[ just like the OGM](https://github.com/OpenVSLAM-Community/openvslam_ros/issues/2#issuecomment-800310426), but I don't believe this is a trivial answer. Maybe you could take a look at [Rtabmap work/paper](http://introlab.github.io/rtabmap/) to have some insights. They use a dense point cloud and a fitting plane technique to detect the floor, but I'm not sure if this technique would work for sparse PC, maybe if your environment has lots of features...

**Reply by Udaykiran87** [reply]
Thank you so much, i will go through it.
